### ISSUE TITLE
Q: Wavetable VCO and one-shot 

### ISSUE THREAD

**Issue by paulpiko** [surge-synthesizer/surge-rack#1024]
Hi. Thanks for the great work you've done. I have some questions. 

If I put a wavetable from WavEditOnline (e.g. PPG_WA07.wav) into the UserWT folder and load it into Wavetable VCO, it is handled as a one-shot. I understand that is because it doesn't have the metadata for clm/cue/smpl.  

I don't want to use the wave as a one-shot.

If I load that same PPG_WA07.wav via Wavetables, Load Untagged Wave as, 256 Sample Frame WaveTable, it is not handled as a one-shot. It works as desired.

The problem with the second approach is I have to manually load each file every time and lose the ability to arrow through as presets. I would like Wavetable VCO to handle a user folder and have it automatically treat each wave as Load Untagged above rather than one-shot. Instead of me doing it manually. Of course that might mean having to define the number of samples somewhere. Or is there a better way to do this?

Alternatively, but not my preference, would be to alter the original files to contain the metadata. I have not found a tool (for Windows) that can easily update a folder of WAV files as needed. Is there a simple way to do it?

**Comment by baconpaul** [comment]
https://github.com/surge-synthesizer/surge/blob/main/scripts/wt-tool/add-surge-metadata.py

that python script will add metadata to the files.

There is no good way to infer correct tags for untagged wavetables, obviously

It sounds like what you want is "if you don't know what the heck to do, assume 256 samples as opposed to one shot" as a user option. Which, of course, will fail for a different group of wavetables.

**Comment by paulpiko** [comment]
Thanks. I've never used python before. I looked at the code and saw the required parameters. I converted a copy of one of the PPG waves like this:

python add-surge-metadata.py --samples 256 PPGMWA00.wav

I then put the output file into a UserWT subfolder and loaded it into a Wavetable VCO. It immediately appeared as shown in the left module in the attached image. It isn't one-shot, that's good. But note that the waveform matches the one on the right which is the bugged waveform after a patch reload, as I reported earlier. So the withsurgemeta file didn't load as desired - did I miss something? 

![image](https://github.com/user-attachments/assets/24de53e6-62e2-414a-8c5e-28dccdffe9e0)

**Comment by baconpaul** [comment]
oh wonder if this is an i16 vs an i15 thing. ok i'll need to actually run it with your wavs and look. that dir has another file which normalizes i16 to i15 wavs which could be it, and surge 1.3 had a few bugs there which may be leaking through?

**Comment by paulpiko** [comment]
When I load the original, unmodified wave through Load Untagged, 256 it loads correctly. There must be something else in that untagged load process that isn't done by the add metadata script. I'll see if I get anywhere with that normalise script. 

Here's an unmodified zip with the wave file from the image in this thread. 
[PPGMWA00.zip](https://github.com/user-attachments/files/18283518/PPGMWA00.zip)

There's also PPG_WA17 attached to the issue about the patch reload problem

**Comment by baconpaul** [comment]
yeah older surge versions had a problem where re-loading an i16 wav file would mis-truncate. I thought i had fixed it (I know I have in the VST) so that's the path of investigation here.

**Comment by paulpiko** [comment]
It looks like running normalise and then add metadata makes a new PPG wav that loads correctly from UserWT. I'll run through the rest of the ~30 waves and see how they go
